pkg/tsdb/cloudmonitoring/cloudmonitoring.go

```diff
@@ -206,6 +206,22 @@ func newInstanceSettings(httpClientProvider httpclient.Provider) datasource.Inst
 	}
 }
 
+func migrateMetricTypeFilter(metricTypeFilter string, prevFilters interface{}) []string {
+	metricTypeFilterArray := []string{"metric.type", "=", metricTypeFilter}
+	filters := []string{}
+	if prevFilters != nil {
+		filtersIface := prevFilters.([]interface{})
+		for _, f := range filtersIface {
+			filters = append(filters, f.(string))
+		}
+		metricTypeFilterArray = append([]string{"AND"}, metricTypeFilterArray...)
+		filters = append(filters, metricTypeFilterArray...)
+	} else {
+		filters = metricTypeFilterArray
+	}
+	return filters
```

Comment:
Nit: if we return early in the both branches of the `if` we don't need a `return` here.